bundles/http/Citrix/Citrix-Virtual-Apps-and-Desktops-service/7687284a-09a1-4494-b33c-f6bfa0932656.js

```diff
@@ -0,0 +1,671 @@
+                }
+                const sessions = response.Items
+
+                if (Array.isArray(sessions) && sessions.length > 0) {
+                    for (const session of sessions) {
+                        flatJson(session, 'Client')
+                        flatJson(session, 'Connection')
+                        flatJson(session, 'User')
+                        flatJson(session, 'Machine')
+                        flatJson(session, 'Machine_DeliveryGroup')
+                        flatJson(session, 'Machine_MachineCatalog')
+                        flatJson(session, 'Machine_Zone')
+
+                        // Store Applications In Use for the Session.
+                        if (session.ApplicationsInUse && session.ApplicationsInUse.length > 0) {
+                            const sessions_applications_in_use = session.ApplicationsInUse.map(
+                                app => {
+                                    app.Session_Id = session.Id
+                                    return app
+                                },
+                            )
+                            params.dataStore.save(
+                                'session_app_in_use',
+                                sessions_applications_in_use,
+                            )
+                        }
+                    }
+                    params.dataStore.save('session', sessions)
+                } else {
+                    console.log(
+                        'WARNING: Endpoint Sessions - Response is empty or in wrong format.',
+                    )
+                }
+            } catch (error) {
+                console.error(error)
+            }
+        } while (continuationToken)
+    }
+
+    console.log('Sync of Sessions finished.')
+}
+
+async function getSiteIds({client, integrationParameters}) {
+    console.log('Started getting Site Ids.')
+
+    let response = await client.fetch('/me', {
+        headers: {
+            'Content-Type': 'application/json',
+            'Citrix-CustomerId': integrationParameters.customerId,
+        },
+    })
+
+    if (!response.ok) {
+        throw new Error(`Getting Site IDs failed(${response.status}: ${response.statusText})`)
+    }
+
+    response = await response.json()
+
+    const siteIds = []
+
+    try {
+        response.Customers.forEach(customer =>
+            customer.Sites.forEach(site => siteIds.push(site.Id)),
+        )
+    } catch (err) {
+        console.error(err)
+    }
+
+    console.log('Finished getting Site IDs')
+
+    return siteIds
+}
+
+async function request(
+    siteId,
+    {client, integrationParameters},
+    entity,
+    continuationToken,
+    fields,
+    body,
+) {
+    let path = `${siteId}/${entity}/$search?fields=${fields}&limit=${limit}`
+
+    if (continuationToken) {
+        path = `${path}&continuationToken=${continuationToken}`
+    }
+    const response = await client.fetch(path, {
+        method: 'POST',
+        body: body,
```

Comment:
```suggestion
        body,
```
When a variable name is the same as the attribute name, then you can use this shorter variant.

Reply:
will do